flink-java/src/main/java/org/apache/flink/api/java/typeutils/GenericTypeInfo.java

```diff
@@ -70,9 +69,6 @@ public boolean isKeyType() {
 
 	@Override
 	public TypeSerializer<T> createSerializer(ExecutionConfig config) {
-		if(config.serializeGenericTypesWithAvro()) {
```

Comment:
POJOs generated by Avro will be handled by the Avro serializer
Generic Types always with Kryo serializer
This change allows to force the POJO serializer to serialize the ENTIRE pojo with Kryo or Avro.

I added a section in the documentation explaining the settings.